### ISSUE TITLE
Proton Experimental does not install prerequisites before launching games.

### ISSUE THREAD

**Issue by ProjectSynchro** [ValveSoftware/Proton#5593]
## System Information
- GPU:  RTX 2070
- Driver/LLVM version:  510.47.03
- Kernel version:  5.16.9
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/ProjectSynchro/f0d4e4f102ca29c71a683ea49dd4df7e
- Proton version: Proton Experimental Specifically: (1645317323 experimental-bleeding-edge-7.0-9804-20220220-pa15ecd-wb3a69b-d80f744-v1cc8af)

I observed this behaviour in all games, some games will launch, but others do not (such as games that require mfc140.dll)

Here is a log of attempting to run Horizon Zero Dawn with Proton Experimental , I picked this as it will not launch at all without it's prerequisites installed beforehand:
[steam-1151640.log](https://github.com/ValveSoftware/Proton/files/8103423/steam-1151640.log)

To make sure I was not running into my own Steam installation causing issues:
- [x] Re-installed Proton Experimental
- [x] Reinstalled Steam (This has fixed issues like this in the past)
- [x] Attempted to use Flatpak Steam (behaviour is still present)

**Comment by ProjectSynchro** [comment]
Appears to be resolved as of updating to Proton Experimental (1645403701 experimental-bleeding-edge-7.0-9892-20220221-pa15ecd-we1bac9-db163a9-v2278da)

Still unsure what went wrong.

**Comment by ProjectSynchro** [comment]
Looks like this has appeared again as of Proton Experimental (1645691642 experimental-bleeding-edge-7.0-10193-20220224-p35a410-wad15cd-db42c07-vedbf49)

Attempted to launch Project Warlock, prerequisites are not installed and the game fails to launch.
[steam-893680.log](https://github.com/ValveSoftware/Proton/files/8132142/steam-893680.log)

Reinstalled Proton Experimental to be sure, this also is observed in other games.

If I should provide other logs or open an issue on https://github.com/ValveSoftware/steam-for-linux/issues instead of here let me know and I will close this again.

**Comment by kisak-valve** [comment]
Hello @ProjectSynchro, as a reference point, does this issue occur with any point releases of Proton or with a full re-install of a game? In general, Proton's bleeding edge builds are automated and untested before they become available to the public.

This reads like the mechanism that transitions the wineprefix to a new version isn't being reliably triggered for trivial commit changes. If that's the case, then this isn't a real issue since the bleeding edge builds are definitely not intended for general, sustained use. Instead, they're intended for temporary tests focused to get quicker feedback on specific topics as adjustments are going into the regular Proton Experimental builds.

**Comment by ProjectSynchro** [comment]
> Hello @ProjectSynchro, as a reference point, does this issue occur with any point releases of Proton or with a full re-install of a game? In general, Proton's bleeding edge builds are automated and untested before they become available to the public.

Makes sense I figured they were automated and very much untested, I use the bleeding-edge builds for semi-regular testing so I had gotten accustomed to using them. I haven't seen this behaviour in any mainline point releases of Proton. That being said I did run into a similar issue a while ago; wiping Steam appeared to have fixed that issue so I suspect there was something wrong with my installation, as I haven't seen this behaviour outside of bleeding-edge builds of Proton Experimental.

> This reads like the mechanism that transitions the wineprefix to a new version isn't being reliably triggered for trivial commit changes. If that's the case, then this isn't a real issue since the bleeding edge builds are definitely not intended for general, sustained use. Instead, they're intended for temporary tests focused to get quicker feedback on specific topics as adjustments are going into the regular Proton Experimental builds.

The behaviour I was seeing was after a clean-install of the game 'Project Warlock', it was also present for prefix updates as well which would make sense if the issue is related to updating/creating the wineprefix. Unfortunately it appears steam has auto-updated Proton Experimental while I was AFK (to 7.0-10220-20220224-p35a410-w690fdd-db42c07-v8a46c2) and the behaviour is no longer present.. I will avoid using bleeding-edge builds for now. The default branch of Proton Experimental and other point releases do not show this issue.

I'll close this issue since it seems this issue is intermittent and may or may not be related to an automated process somewhere, as some builds seem to work correctly and some do not. (May be related to whatever build machines are being used etc, I can only guess...)

Cheers.
